src/zope/app/form/browser/widgets.rst

```diff
@@ -13,56 +13,60 @@ description
 Simple stuff.
 
 Our Poll product holds an editable list of the `PollOption` instances.
-This is shown in the ``poll.py`` source below::
 
-    from persistent import Persistent
-    from interfaces import IPoll, IPollOption
-    from zope.interface import implements, classImplements
+First the Schemas are defined in the ``interfaces.py`` file below::
+
+    >>> from zope.interface import Interface
+    >>> from zope.schema import Object, Tuple, TextLine
+    >>> from zope.schema.interfaces import ITextLine
+    >>> from zope.i18nmessageid import MessageFactory
+
+    >>> _ = MessageFactory("poll")
+
+    >>> class IPollOption(Interface):
+    ...    label = TextLine(title=u'Label', min_length=1)
+    ...    description = TextLine(title=u'Description', min_length=1)
+
+    >>> class IPoll(Interface):
+    ...    options = Tuple(title=u'Options',
+    ...        value_type=Object(IPollOption, title=u'Poll Option'))
+    ...
+    ...    def getResponse(option): "get the response for an option"
+    ...
+    ...    def choose(option): 'user chooses an option'
+
+The implementation is shown in the ``poll.py`` source below::
+
+    >>> from persistent import Persistent
+    >>> from zope.interface import implementer, classImplements
+
+    >>> @implementer(IPollOption)
+    ... class PollOption(Persistent, object):
+    ...       pass
+
+    >>> @implementer(IPoll)
+    ... class Poll(Persistent, object):
+    ...
+    ...
+    ...  def getResponse(self, option):
+    ...        return self._responses[option]
+    ...
+    ...  def choose(self, option):
+    ...        self._responses[option] += 1
+    ...        self._p_changed = 1
+    ...
+    ...  def get_options(self):
+    ...        return self._options
+    ...
+    ...  def set_options(self, options):
+    ...        self._options = options
+    ...        self._responses = {}
+    ...        for option in self._options:
+    ...            self._responses[option.label] = 0
+    ...
+    ...  options = property(get_options, set_options, None, 'fiddle options')
```

Comment:
FWIW my reviews are generally "here's how I think you can make the code better", and not "you must do this before I'll allow a merge".  I'm bad at stating that explicitly (and I'm bad at using [positive language](https://www.pagerduty.com/blog/better-code-reviews-empathy/) in my reviews, despite my frequent resolutions to get better at it).